editor/modes/text-editor/index.js

```diff
@@ -5,55 +5,106 @@ import { connect } from 'react-redux';
 import Textarea from 'react-autosize-textarea';
 
 /**
+ * WordPress dependencies
+ */
+import { Component } from 'element';
+import { serialize, parse } from 'blocks';
+
+/**
  * Internal dependencies
  */
 import './style.scss';
 import PostTitle from '../../post-title';
 import { getBlocks } from '../../selectors';
 
-function TextEditor( { blocks, onChange } ) {
-	return (
-		<div className="editor-text-editor">
-			<header className="editor-text-editor__formatting">
-				<div className="editor-text-editor__formatting-group">
-					<button className="editor-text-editor__bold">b</button>
-					<button className="editor-text-editor__italic">i</button>
-					<button className="editor-text-editor__link">link</button>
-					<button>b-quote</button>
-					<button className="editor-text-editor__del">del</button>
-					<button>ins</button>
-					<button>img</button>
-					<button>ul</button>
-					<button>ol</button>
-					<button>li</button>
-					<button>code</button>
-					<button>more</button>
-					<button>close tags</button>
+class TextEditor extends Component {
+	constructor( { blocks } ) {
+		super( ...arguments );
+		this.state = {
+			blocks,
+			value: serialize( blocks ),
+		};
+		this.onChange = this.onChange.bind( this );
+		this.onBlur = this.onBlur.bind( this );
+	}
+
+	onChange( event ) {
+		this.setState( {
+			value: event.target.value,
+		} );
+		this.props.markDirty();
+	}
+
+	onBlur() {
+		const blocks = parse( this.state.value );
```

Comment:
As noted in chat, can we test here whether the value has changed from what we know to have been the previous value before parsing and assigning new blocks?